ribs/schedulers/_bandit_scheduler.py

```diff
@@ -0,0 +1,325 @@
+                             "defaults to be the same as `archive` if you pass "
+                             "`result_archive=None`")
+
+        self._archive = archive
+        self._emitter_pool = np.array(emitter_pool)
+        self._num_active = num_active
+        self._add_mode = add_mode
+        self._result_archive = result_archive
+        self._reselect = reselect
+
+        # Boolean mask of the active emitters. Initializes to the first
+        # num_active emitters in the emitter pool.
+        self._active_arr = np.zeros_like(self._emitter_pool, dtype=bool)
+        self._active_arr[:self._num_active] = True
+
+        # Used by UCB1 to select emitters.
+        self._success = np.zeros_like(self._emitter_pool, dtype=float)
+        self._selection = np.zeros_like(self._emitter_pool, dtype=float)
+        self._restarts = np.zeros_like(self._emitter_pool, dtype=int)
+        self._zeta = zeta
+
+        # Keeps track of whether the scheduler should be receiving a call to
+        # ask() or tell().
+        self._last_called = None
+        # The last set of solutions returned by ask().
+        self._solution_batch = []
+        # The number of solutions created by each emitter.
+        self._num_emitted = np.array([None for _ in self._active_arr])
+
+    @property
+    def archive(self):
+        """ribs.archives.ArchiveBase: Archive for storing solutions found in
+        this scheduler."""
+        return self._archive
+
+    @property
+    def emitters(self):
+        """list of ribs.archives.EmitterBase: Emitters for generating solutions
+        in this scheduler."""
+        return self._active_arr
+
+    @property
+    def result_archive(self):
+        """ribs.archives.ArchiveBase: Another archive for storing solutions
+        found in this optimizer.
+        If `result_archive` was not passed to the constructor, this property is
+        the same as :attr:`archive`.
+        """
+        return (self._archive
+                if self._result_archive is None else self._result_archive)
+
+    def ask(self):
+        """Generates a batch of solutions by calling ask() on all emitters.
+
+        The emitters used by ask are determined by the UCB1 algorithm. Briefly,
+        emitters that have never been selected are prioritized, then emitters
+        are sorted in descending order based the accurary of their past
+        prediction.
+
+        .. note:: The order of the solutions returned from this method is
+            important, so do not rearrange them.
+
+        Returns:
+            (batch_size, dim) array: An array of n solutions to evaluate. Each
+            row contains a single solution.
+        Raises:
+            RuntimeError: This method was called without first calling
+                :meth:`tell`.
+        """
+        if self._last_called in ["ask", "ask_dqd"]:
+            raise RuntimeError("ask_dqd cannot be called immediately after " +
```

Comment:
`ask`? You should also add an `ask_dqd` and `tell_dqd` that throw an error because they are not supported

Reply:
the error should be "NotImplementedError("some message")"